packages/react-reconciler/src/ReactFiberTransition.new.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+import type {FiberRoot} from './ReactInternalTypes';
+import type {Lanes} from './ReactFiberLane.new';
+import type {StackCursor} from './ReactFiberStack.new';
+import type {Cache, SpawnedCachePool} from './ReactFiberCacheComponent.new';
+
+import {enableCache} from 'shared/ReactFeatureFlags';
+import {isPrimaryRenderer} from './ReactFiberHostConfig';
+import {createCursor, push, pop} from './ReactFiberStack.new';
+import {getWorkInProgressRoot} from './ReactFiberWorkLoop.new';
+import {
+  createCache,
+  retainCache,
+  CacheContext,
+} from './ReactFiberCacheComponent.new';
+
+// When retrying a Suspense/Offscreen boundary, we restore the cache that was
+// used during the previous render by placing it here, on the stack.
+const resumedCache: StackCursor<Cache | null> = createCursor(null);
+
+function peekCache(): Cache | null {
+  if (!enableCache) {
+    return (null: any);
+  }
+
+  // Check if the cache pool already has a cache we can use.
+
+  // If we're rendering inside a Suspense boundary that is currently hidden,
+  // we should use the same cache that we used during the previous render, if
+  // one exists.
+  const cacheResumedFromPreviousRender = resumedCache.current;
+  if (cacheResumedFromPreviousRender !== null) {
+    return cacheResumedFromPreviousRender;
+  }
+
+  // Otherwise, check the root's cache pool.
+  const root = (getWorkInProgressRoot(): any);
+  const cacheFromRootCachePool = root.pooledCache;
+
+  return cacheFromRootCachePool;
+}
+
+export function requestCache(renderLanes: Lanes): Cache {
```

Comment:
"pool" still makes sense here because that's where this function requests the cache from. So I would rename this one back to `requestCacheFromPool`, and `peekCache` back to `peekCacheFromPool`.